index.js

```diff
@@ -1,38 +1,89 @@
+//final week 10
 import express from 'express'
 import bodyParser from 'body-parser'
 import redis from 'redis'
 
-// make the one and only express object     -   done?
 const app = express()
 
-// make a database connection to redis      -   done?
 var urlencodedParser = bodyParser.urlencoded({ extended: false})
 const db = redis.createClient({
     url: 'redis://localhost:6379'
 })
-await db.connect()     // if this errors, make sure redis is running    -   done?
+await db.connect()    
 
-// add routes
-// this route serves static files
 app.use(express.static('public'))
 
-// this route serves the home page, using code
-app.post('/add', urlencodedParser, async (req, res) => {
-    await db.set("form-data", req.body.answerText)
-    res.send(`<!DOCTYPE html><html><body>Data saved to Redis key named 'form-data'</body></html>`)
-    res.redirect()
+
+
+app.get('/', async(req, res) => {
+    res.send(`<!DOCTYPE html><html><body><A href="/home.html">Say Hello</A></body></html>`)
 })
+
+app.post('/add', urlencodedParser, async(req, res) => {
+    var quizName = req.body.quizName
+    await db.set('quizName', quizName)
```

Comment:
Because many people will use the server at the same time, there are many quizes and many users taking quizes at the same time. If the quizName key is just `quizName`, then the users will overwrite each other's data.

See the handout for the format string to use. It is given in the assignment on week 10 (and again in week 11).

Reply:
Sorry, I just noticed these comments. How can I fix this?

Reply:
Remove the code that is writing quizName to redis, as well as the incr of 'tttt'.

See the comment in the other handler on how to get quizName from the url.